### PR TITLE
Update DG for `undo` command

### PR DESCRIPTION
**Pull request by rexcyrio**

### CODE REVIEW

File: docs/images/UndoRedoState5.png
Comment (minosx31) [inline docs/images/UndoRedoState5.png:1]: Maybe we can remove this image since it is not used anymore?

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -187,82 +187,122 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+   <figure style="text-align: center;">
 
-<img src="images/CommitActivityDiagram.png" width="250" />
+   ![UndoRedoState3](images/UndoRedoState3.png)
+
+   <figcaption>Figure 4: State of the address book history after calling <code>undo</code></figcaption>
+
+   </figure>
+
+1. The user then decides to execute the command `list`. Commands that do not modify the address book (such as `list`) will usually not call `Model#commit()` or `Model#undo()`. Thus, the `addressBookStateList` remains unchanged.
```
Comment (minosx31) [inline docs/DeveloperGuide.md:265]: Nice to have included this because it allows readers to better understand the `undo` feature!

**Review by minosx31 — COMMENTED** [review]
Are you planning to implement `redo` feature in future? If not, maybe it will be better to rename the diagrams and images to not include `redo`.

File: docs/images/UndoRedoState5.png
Comment (rexcyrio) [reply docs/images/UndoRedoState5.png:1]: Good catch!

**Review by niekis — APPROVED** [review]
LGTM!

**Pull request merged**